p-token/src/processor/shared/initialize_mint.rs

```diff
@@ -55,24 +56,24 @@ pub fn process_initialize_mint(
         (mint_info, None)
     };
 
-    // SAFETY: single mutable borrow to `mint_info` account data.
-    let mint = unsafe { load_mut_unchecked::<Mint>(mint_info.borrow_mut_data_unchecked())? };
-
-    if mint.is_initialized() {
-        return Err(TokenError::AlreadyInUse.into());
-    }
-
-    // Check rent-exempt status of the mint account.
+    let mint_data_len = mint_info.data_len();
```

Comment:
This worried me for a second, but I confirm that `load_mut_unchecked` does a strict check of the account length, so this is all good

Reply:
Yeah, this surprised me a bit as well. I checked that current SPL Token does the same thing: https://github.com/solana-program/token/blob/main/program/src/processor.rs#L36